src/EasyAbp.BookingService.Application/EasyAbp/BookingService/PeriodSchemes/PeriodSchemeAppService.cs

```diff
@@ -116,15 +116,25 @@ public virtual async Task<PeriodSchemeDto> DeletePeriodAsync(Guid periodSchemeId
     {
         var periodScheme = await GetEntityByIdAsync(periodSchemeId);
 
-        var period = periodScheme.Periods.Single(x => x.Id == periodId);
-
-        periodScheme.Periods.Remove(period);
+        await _periodSchemeManager.DeletePeriodAsync(periodScheme, periodId);
 
         await _repository.UpdateAsync(periodScheme, true);
 
         return await MapToGetOutputDtoAsync(periodScheme);
     }
 
+    public override async Task DeleteAsync(Guid id)
+    {
+        await CheckDeletePolicyAsync();
+        var entity = await GetEntityByIdAsync(id);
+        if (await _periodSchemeManager.IsPeriodSchemeInUseAsync(entity))
+        {
+            throw new CannotDeletePeriodSchemeInUseException(entity.Id, entity.Name);
+        }
+
+        await DeleteByIdAsync(id);
```

Comment:
Better use `await _repository.DeleteAsync(entity, true);` to avoid duplicate getting entity:
```CSharp
    public virtual async Task DeleteAsync(TKey id, bool autoSave = false, CancellationToken cancellationToken = default)
    {
        var entity = await FindAsync(id, cancellationToken: cancellationToken);
        if (entity == null)
        {
            return;
        }

        await DeleteAsync(entity, autoSave, cancellationToken);
    }
```